File: app/src/main/java/kyklab/burner2/fm/FMAdapter.java
package kyklab.burner2.fm;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.io.File;
import java.util.List;

import kyklab.burner2.R;

public class FMAdapter extends RecyclerView.Adapter<FMAdapter.ViewHolder> {
    private final Context pContext;
    private final FMAdapterCallback mCallback;
    private final List<File> mFileList;

    public FMAdapter(Context pContext, List<File> mFileList) {
        this.pContext = pContext;
        this.mFileList = mFileList;
        this.mCallback = (FMAdapterCallback) pContext;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_file, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        File file = mFileList.get(position);
        if (file.isDirectory()) {
            Glide.with(pContext).clear(holder.fileIconView);
            holder.fileIconView.setImageResource(R.drawable.ic_folder_36dp);
        } else {
            Glide.with(pContext).load(file).centerCrop()
                    .into(holder.fileIconView);
        }
        holder.fileNameView.setText(file.getName());
    }

    @Override
    public int getItemCount() {
        return mFileList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        private final ImageView fileIconView;
        private final TextView fileNameView;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            fileIconView = itemView.findViewById(R.id.fileIconView);
            fileNameView = itemView.findViewById(R.id.fileNameView);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    File file = mFileList.get(getAdapterPosition());
                    if (file.isDirectory()) {
                        mCallback.enterDirectory(file.getName());
                    } else {
                        mCallback.customPictureSelected(file.getAbsolutePath());
                    }
                }
            });
        }
    }
}

File: app/src/main/java/kyklab/burner2/fm/FMAdapter.kt
package kyklab.burner2.fm

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import kyklab.burner2.R
import java.io.File

class FMAdapter(private val pContext: Context, private val mFileList: List<File>) : RecyclerView.Adapter<FMAdapter.ViewHolder>() {
    private val mCallback: FMAdapterCallback

    init {
        this.mCallback = pContext as FMAdapterCallback
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_file, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val file = mFileList[position]
        if (file.isDirectory) {
            Glide.with(pContext).clear(holder.fileIconView)
            holder.fileIconView.setImageResource(R.drawable.ic_folder_36dp)
        } else {
            Glide.with(pContext).load(file).centerCrop()
                    .into(holder.fileIconView)
        }
        holder.fileNameView.text = file.name
    }

    override fun getItemCount(): Int {
        return mFileList.size
    }

    internal inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val fileIconView: ImageView
        private val fileNameView: TextView

        init {
            fileIconView = itemView.findViewById(R.id.fileIconView)
            fileNameView = itemView.findViewById(R.id.fileNameView)

            itemView.setOnClickListener {
                val file = mFileList[adapterPosition]
                if (file.isDirectory) {
                    mCallback.enterDirectory(file.name)
                } else {
                    mCallback.selectAsCustomPicture(file.absolutePath)
                }
            }
        }
    }
}
